crates/neon/src/macro_internal/mod.rs

```diff
@@ -11,3 +16,55 @@ pub static EXPORTS: [for<'cx> fn(&mut ModuleContext<'cx>) -> NeonResult<Export<'
 
 #[linkme::distributed_slice]
 pub static MAIN: [for<'cx> fn(ModuleContext<'cx>) -> NeonResult<()>];
+
+pub trait NeonExportReturnValue<'cx> {
```

Comment:
Let's put a comment either here or on the `impl … for &T` explaining how this leverages the Serde signature + autoref specialization.

Reply:
Done. I added some documentation describing how this works.